workflows/readmapping.nf

```diff
@@ -125,14 +131,31 @@ workflow READMAPPING {
     ALIGN_ONT ( PREPARE_GENOME.out.fasta, ch_reads.ont )
     ch_versions = ch_versions.mix ( ALIGN_ONT.out.versions )
 
-
+    // gather alignments
     ch_aligned_bams = Channel.empty()
     | mix( ALIGN_HIC.out.bam )
     | mix( ALIGN_ILLUMINA.out.bam )
     | mix( ALIGN_HIFI.out.bam )
     | mix( ALIGN_CLR.out.bam )
     | mix( ALIGN_ONT.out.bam )
-    CONVERT_STATS ( ch_aligned_bams, PREPARE_GENOME.out.fasta )
+
+    // Optionally insert params.header information to bams
+    ch_reheadered_bams = Channel.empty()
+    if ( params.header ) {
+        ch_combined = ch_aligned_bams.map { meta, bam, _ ->
+            def suffix = bam instanceof List ? bam[0].getExtension() : bam.getExtension()
+            meta.suffix = suffix // add suffix to meta so output matches input type
+            [meta, bam, file( params.header )]
```

Comment:
Check the lines 9-15 of this file. File parameters, if defined, are first checked with `checkIfExists: true` and then turned into channels outside of the `workflow { ... }` block.
Assuming you move the suffix selection to the module and we don't need to deal with the padding, I feel calling `SAMTOOLS_REHEADER` could be simplified to something like:
```
SAMTOOLS_REHEADER( ch_aligned_bams, ch_header.first() )
```